### ISSUE TITLE
interleaved reads

### ISSUE THREAD

**Issue by macmanes** [Blahah/transrate#162]
Can we have an option of inputting interleaved reads, like from khmer/seqtk? Not sure if SNAP/salmon can handle this..

```
transrate --assembly bwa.Trinity.fasta --interleaved Schizo.50M.interleaved.fq.gz --threads 16
```

**Comment by rob-p** [comment]
Hi @macmanes,

  It looks like SNAP has some code for this, but I'm not sure how well tested it is.  However, assuming that SNAP handles it and generates a valid BAM, Salmon is indifferent on how the raw reads were provided to the aligner.  That is, Transrate talks to Salmon only through BAM, so if SNAP handles it, it Salmon should be fine.

**Comment by macmanes** [comment]
OK - cool!

**Comment by Blahah** [comment]
Yeah SNAP can handle it, so we just need to add it to the Transrate interface. How about if left and right are the same file, assume interleaved? I'd prefer to avoid adding another argument.

**Comment by macmanes** [comment]
might be asking for trouble with a non-standard CLI option.. I do get your point about not adding more options tho..

**Comment by Blahah** [comment]
Interleaving is non-standard IMO (grumble grumble...). Are you thinking that the 'standard' would be to have a third argument, `--interleaved` or something similar?

**Comment by macmanes** [comment]
yes, `--interleaved` is what I was thinking about.

**Comment by ctb** [comment]
+1

Interleaved supports streaming :>

**Comment by ctb** [comment]
(What the heck does "non-standard" mean in this field, anyway?)

**Comment by Blahah** [comment]
Just that very few tools use it. khmer is the only thing I use frequently that does I think. But, users are in charge, so we'll add it :)
